src/main/java/build/buildfarm/cas/cfc/CASFileCache.java

```diff
@@ -393,7 +397,7 @@ private boolean contains(
         result.mergeFrom(digest).setSizeBytes(entry.size);
       }
       onContains.accept(key);
-      return true;
+      return blobExistsOnLocalStorage(key);
```

Comment:
To prevent the extra overhead, can we only perform the check if ensure outputs present is set?

Reply:
that's a good idea.  I'll adjust for that.